src/transformers/models/imagebind/processing_imagebind.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2024 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Image/Text processor class for ImageBind
+"""
+
+from typing import List, Optional, Union
+
+import numpy as np
+
+
+try:
+    from typing import Unpack
+except ImportError:
+    from typing_extensions import Unpack
+
+from ...image_utils import (
+    ImageInput,
+    VideoInput,
+)
+from ...processing_utils import AudioKwargs, ImagesKwargs, ProcessingKwargs, ProcessorMixin
+from ...tokenization_utils_base import (
+    BatchEncoding,
+    PreTokenizedInput,
+    TextInput,
+)
+
+
+class ImageBindProcessorImagesKwargs(ImagesKwargs, total=False):
+    do_convert_rgb: bool
+    do_chunk: bool
+    chunk_duration: float
+    num_chunks: int
+    num_frames_per_chunk: int
+    fps: int
```

Comment:
As per the doc of ProcessorKwargs only the kwargs which are new are to be added like this and as only some of them are not in the default `ImagesKwargs` so rest of them can be removed

Reply:
Update:made necessary changes in the recent commits

Reply:
:smiling_face_with_tear: Oh! I see, you meant the type hinting of `do_convert_rgb`. Well, in `image_processing` file it is not optional [here](https://github.com/huggingface/transformers/blob/f878996feac659a8e93701b4592e2329d54613f9/src/transformers/models/imagebind/image_processing_imagebind.py#L334) and is being used in `_preprocess` so that's why it wasn't kept `Optional` here too